### DISCUSSION TITLE
2d liddriven cavity problem

### DISCUSSION THREAD

**Discussion by ztdepztdep** [CFD-GO/TCLB#337] [Q&A] [answered]
```
<?xml version="1.0"?>
<CLBConfig version="2.0" output="output/" permissive="true">
	<Geometry nx="100" ny="100">
		<BGK>			<Box/>		</BGK>
		<WVelocity name="Lid">	<Box ny="-1"/>
        </WVelocity>
		<Wall>
			<Box nx="1"/> <Box nx="-1"/>
			<Box ny="1"/> <Box ny="-1"/>
		</Wall>
	</Geometry>
	<Model>
		<Param name="Velocity" value="1.0"/>
		<Param name="nu" value="0.001"/>
	</Model>
	<VTK Iterations="5000"/>
	<Solve Iterations="50000"/>
</CLBConfig>
```
I wrote a 2D lid-driven cavity problem,  the results give a zero velocity field. Could you please give me some help?

**Comment by mdzik** [comment]
I would say It's more likely NaN. Those should be Lattice Units (see lbunit.pdf in google, I will try to provide legitimate link)

Try changing Velocity to 0.01 (max around 0.1)
And nu to 0.16666

Add second VTK/Solve block before one you have
```xml
	<Solve Iterations="500">
	    <VTK Iterations="50"/>
        </Solve>
```
To see start-up of the simulation (check if it's not crashing after few steps)

**Comment by llaniewski** [comment]
Please check if the model you are using the name of the setting is "Velocity" and not "VelocityX" or something else. You can see all the details of the model in `CLB/[model name]/SUMMARY` file, or for standard models in the [documentation](https://docs.tclb.io/models/flow/d2q9/d2q9/).

**Comment by ztdepztdep** [comment]
> I would say It's more likely NaN. Those should be Lattice Units (see lbunit.pdf in google, I will try to provide legitimate link)
> 
> Try changing Velocity to 0.01 (max around 0.1)
> And nu to 0.16666
> 
> Add second VTK/Solve block before one you have
> 
> ```
> 	<Solve Iterations="500">
> 	    <VTK Iterations="50"/>
>         </Solve>
> ```
> 
> To see start-up of the simulation (check if it's not crashing after few steps)

Thanks, I want to model the flow at Re=1000, so nu=0.001.

**Comment by llaniewski** [comment, marked as answer]
@ztdepztdep Remember that TCLB is a research code. It is **not a general purpose CFD code**.

If you want to calculate lid driven cavity at Re=1000, you either need a very big mesh, or more stable model then SRT. You also have to use (or potentially implement) a good boundary condition for the lid. The `WVelocity` is "west" velocity Zou-He boundary condtion. This means it can be used only on the "west" (left) edge of the domain (it fills the missing f1 f5 f8 densities).

**Reply by ztdepztdep** [reply]
![Screenshot_20210901_215015](https://user-images.githubusercontent.com/8356693/131683601-bc754cd4-97d2-456a-912b-3d8f176a9a3c.png)

 
Could you please give me some information about how to adjust the Reynolds number. I have tried to change the "viscorsity" parameter, but it seems the results doesn't change as shown in the picture.
